File: JAVA/1957. Delete Characters to Make Fancy String.java
class Solution {
    public String makeFancyString(String s) {
        if(s.length() < 2) return s;
        StringBuilder k = new StringBuilder();
        k.append(s.charAt(0)).append(s.charAt(1));
        for(int x = 2; x < s.length(); x++) {
            if(s.charAt(x) == s.charAt(x - 1) && s.charAt(x) == s.charAt(x - 2)) continue;
            k.append(s.charAt(x));
        }
        return k.toString();
    }
}

File: C++/1957. Delete Characters to Make Fancy String.cpp
class Solution {
public:
    string makeFancyString(string s) {
        if(s.size()<3)
        return s;
        for(int i=s.size()-1;i>=2;i--){
            if(s[i]==s[i-1]){
                if(s[i-1]==s[i-2])
                s.erase(s.begin()+i);
            }
        }
        return s;
    }
};